.ci/ansible/playbook.yml

```diff
@@ -30,7 +30,7 @@
   pre_tasks:
     - name: Update apt cache.
       apt: update_cache=yes cache_valid_time=600
-      when: ansible_os_family == 'Debian'
+      when: ansible_pkg_mgr == 'apt' or ansible_distribution in ["Debian", "Ubuntu"]
```

Comment:
IIUC, `apt` is the only package manager for `Debian` specific (including Ubuntu). So what's the reason for being so explicit?

Reply:
The error comes from https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/PR-2103/3/pipeline, where a Debian machine (as declared in https://github.com/elastic/e2e-testing/blob/main/.ci/.e2e-tests.yaml#L4) hypothetically with Debian 10:
<img width="541" alt="Screenshot 2022-02-08 at 10 28 14" src="https://user-images.githubusercontent.com/951580/152957836-d00a821e-4924-48f2-8777-824056055a9a.png">
 
does not find the gnugp package because the pre_task to update cache is skipped

<img width="1736" alt="Screenshot 2022-02-08 at 10 30 25" src="https://user-images.githubusercontent.com/951580/152958287-dbf27064-ab00-4e70-a1bf-0e2bac88124d.png">

On the contrary, I've seen builds where the pre_task to update cache is in fact called:

<img width="1790" alt="Screenshot 2022-02-08 at 10 31 33" src="https://user-images.githubusercontent.com/951580/152958493-2d301bfb-5923-4396-8f79-bb466df5cca6.png">
 
So my guess is the previous Ansible code is not evaluated

Reply:
Gotcha, I understand the issue, but I don't understand the condition yet, maybe `when: ansible_pkg_mgr == 'apt'` should be sufficient?

Reply:
I'm pretty sure it is. I simply copied the very same when condition we have in https://github.com/elastic/e2e-testing/blob/main/.ci/ansible/tasks/install_deps.yml#L14

We could simplify all in this PR though

Reply:
Done in https://github.com/elastic/e2e-testing/pull/2107/commits/26037f0a593922f3d2f7defffbffa77fcb0ce052